scheduler/remotesync.go

```diff
@@ -14,180 +14,114 @@
-}
+// IDs return the complete list of service ids.
+// Implements utils.Synchronizer
+func (sync *RemoteServiceSync) IDs() ([]string, error) {
+	svcs, err := sync.iface.GetServices(sync.poolID)
+	if err != nil {
+		glog.Errorf("Could not look up services in pool %s: %s", sync.poolID, err)
+		return nil, err
+	}
 
-func (s *scheduler) GetResourcePools() ([]pool.ResourcePool, error) {
-	return s.facade.GetResourcePools(datastore.Get())
+	ids := make([]string, len(svcs))
+	for i, svc := range svcs {
+		ids[i] = svc.ID
+	}
+	return ids, nil
 }
 
-func (s *scheduler) AddUpdateResourcePool(pool *pool.ResourcePool) error {
-	if p, err := s.facade.GetResourcePool(datastore.Get(), pool.ID); err != nil {
+// Create creates a new service.
+// Implements utils.Synchronizer
+func (sync *RemoteServiceSync) Create(data interface{}) error {
+	svc, err := sync.convert(data)
+	if err != nil {
 		return err
-	} else if p == nil {
-		return s.facade.AddResourcePool(datastore.Get(), pool)
 	}
-
-	return s.facade.UpdateResourcePool(datastore.Get(), pool)
-}
-
-func (s *scheduler) RemoveResourcePool(id string) error {
-	return s.facade.RemoveResourcePool(datastore.Get(), id)
-}
-
-func (s *scheduler) GetServicesByPool(id string) ([]service.Service, error) {
-	return s.facade.GetServicesByPool(datastore.Get(), id)
+	return sync.iface.AddService(svc)
 }
 
-func (s *scheduler) AddUpdateService(svc *service.Service) error {
-	if sv, err := s.facade.GetService(datastore.Get(), svc.ID); err != nil {
+// Update updates existing service.
+// Implements utils.Synchronizer
+func (sync *RemoteServiceSync) Update(data interface{}) error {
+	svc, err := sync.convert(data)
+	if err != nil {
 		return err
-	} else if sv == nil {
-		return s.facade.AddService(datastore.Get(), *svc)
 	}
-
-	return s.facade.UpdateService(datastore.Get(), *svc)
+	return sync.iface.UpdateService(svc)
 }
 
-func (s *scheduler) RemoveService(id string) error {
-	return s.facade.RemoveService(datastore.Get(), id)
+// Delete deletes an existing service
+// Implements utils.Synchronizer
+func (sync *RemoteServiceSync) Delete(id string) error {
+	return sync.iface.RemoveService(id)
 }
 
-func (s *scheduler) GetHostsByPool(id string) ([]host.Host, error) {
-	return s.facade.FindHostsInPool(datastore.Get(), id)
-}
-
-func (s *scheduler) AddUpdateHost(host *host.Host) error {
-	if h, err := s.facade.GetHost(datastore.Get(), host.ID); err != nil {
-		return err
-	} else if h == nil {
-		return s.facade.AddHost(datastore.Get(), host)
+// convert transforms an interface into domain service object
+func (sync *RemoteServiceSync) convert(data interface{}) (*service.Service, error) {
+	if svc, ok := data.(*service.Service); ok {
+		return svc, nil
 	}
-
-	return s.facade.UpdateHost(datastore.Get(), host)
-}
-
-func (s *scheduler) RemoveHost(id string) error {
-	return s.facade.RemoveHost(datastore.Get(), id)
-}
+	return nil, errors.New("invalid type")
```

Comment:
Just nitpicking, but maybe use a more expressive string here? I dunno, "Could not convert data to Service object" or something?